prow/jenkins/metrics.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jenkins
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	requests = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "jenkins_requests",
+		Help: "Number of Jenkins requests made from prow.",
+	}, []string{
+		// http verb of the request
+		"verb",
+		// path of the request
+		"handler",
+		// http status code of the request
+		"code",
+	})
+	requestRetries = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "jenkins_request_retries",
+		Help: "Number of Jenkins request retries made from prow.",
+	})
+	requestLatency = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "jenkins_request_latency",
+		Help:    "Time for a request to roundtrip between prow and Jenkins.",
+		Buckets: prometheus.DefBuckets,
+	}, []string{
+		// http verb of the request
+		"verb",
+		// path of the request
+		"handler",
+	})
+	resyncPeriod = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name:    "resync_period_seconds",
+		Help:    "Time the controller takes to complete one reconcilation loop.",
+		Buckets: prometheus.ExponentialBuckets(1, 3, 5),
+	})
+	builds = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "jenkins_builds",
+		Help: "Number of Jenkins builds reconciled by prow.",
+	}, []string{
+		// state of the build: enqueued, running, success, failure, aborted, all
```

Comment:
@smarterclayton is there any prom query that can give me the count of all builds in all states so I can drop `all` which feels too hacky?